### PR TITLE
Neural Machine Translation with Attention

### PR DESCRIPTION
**Pull request by yashk2810**
Added Neural Machine Translation with Attention notebook

### CODE REVIEW

**Comment by yashk2810** [comment]
@alextp

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb [hunk 1]
```diff
@@ -0,0 +1,992 @@
...
+        "colab": {
+          "autoexec": {
+            "startup": false,
+            "wait_interval": 0
+          }
+        }
+      },
+      "cell_type": "code",
+      "source": [
+        "# first we remove the pronumciations\n",
```
Comment (alextp) [inline tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb]: pronunciations

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb [hunk 2]
```diff
@@ -0,0 +1,992 @@
...
+        "class Decoder(tf.keras.Model):\n",
+        "    def __init__(self, vocab_size, embedding_dim, dec_units, batch_sz):\n",
+        "        super(Decoder, self).__init__()\n",
+        "        self.batch_sz = batch_sz\n",
+        "        self.dec_units = dec_units\n",
+        "        self.embedding = tf.keras.layers.Embedding(vocab_size, embedding_dim)\n",
+        "        \n",
+        "        # If you have a GPU, we recommend using CuDNNGRU(provides a 3x speedup than GRU)\n",
+        "        # the code automatically does that.\n",
+        "        if tf.test.is_gpu_available():\n",
```
Comment (alextp) [inline tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb]: Refactor this into a GRU function so you can remove the duplicates?

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb [hunk 3]
```diff
@@ -0,0 +1,992 @@
...
+        "            dec_hidden = enc_hidden\n",
+        "            \n",
+        "            dec_input = tf.expand_dims([targ_lang.word2idx['<start>']] * BATCH_SIZE, 1)       \n",
+        "            \n",
+        "            # Teacher forcing - feeding the target as the next input\n",
+        "            for t in range(1, targ.shape[1]):\n",
+        "                # passing enc_output to the decoder\n",
+        "                predictions, dec_hidden, _ = decoder(dec_input, dec_hidden, enc_output)\n",
+        "                \n",
+        "                loss += loss_function(targ[:, t], predictions)\n",
```
Comment (alextp) [inline tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb]: I think you also need to worry here about sequences of different lengths, by masking the loss

**Review by alextp — CHANGES_REQUESTED** [review]

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb (hunk 1 above)
Comment (yashk2810) [reply tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb]: Done

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb (hunk 2 above)
Comment (yashk2810) [reply tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb]: Done

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb (hunk 3 above)
Comment (yashk2810) [reply tensorflow/contrib/eager/python/examples/nmt_with_attention/NMT_with_Attention.ipynb]: Done

**Review by alextp — APPROVED** [review]
